core/base/compactTriangulation/CompactTriangulation.h

```diff
@@ -0,0 +1,4214 @@
+          }
+
+          bool error = false;
+
+#ifdef TTK_ENABLE_OPENMP
+#pragma omp parallel for num_threads(this->threadNumber_)
+#endif
+          for(SimplexId i = 0; i < cellNumber; i++) {
+            if(offset[i + 1] - offset[i] - 1 != cellDimension) {
+#ifdef TTK_ENABLE_OPENMP
+#pragma omp atomic write
+#endif // TTK_ENABLE_OPENMP
+              error = true;
+            }
+          }
+
+          if(error) {
+            this->printErr("Unable to create triangulation for "
+                           "inhomogeneous\ncell dimensions.");
+            return -2;
+          }
+        }
+      }
+
+      if(cellNumber_)
+        clear();
+
+      cellNumber_ = cellNumber;
+      vector<SimplexId> vertexMap(vertexNumber_);
+      reorderVertices(vertexMap);
+      reorderCells(vertexMap, cellNumber, connectivity, offset);
+      cellArray_
+        = std::make_shared<CellArray>(connectivity, offset, cellNumber);
+
+      // ASSUME Regular Mesh Here to compute dimension!
+      if(cellNumber) {
+        if(cellArray_->getCellVertexNumber(0) == 3) {
+          maxCellDim_ = 2;
+        } else {
+          maxCellDim_ = 3;
+        }
+      }
+
+      return 0;
+    }
+#else
+    // Flat layout with a single array (legacy & default one)
+    inline int setInputCells(const SimplexId &cellNumber,
+                             const LongSimplexId *cellArray) {
+      if(cellNumber_)
+        clear();
+
+      cellNumber_ = cellNumber;
+
+      if(cellNumber) {
+        // assume regular mesh here to compute dimension
+        maxCellDim_ = cellArray[0] - 1;
+        vector<SimplexId> vertexMap(vertexNumber_);
+        reorderVertices(vertexMap);
+        reorderCells(vertexMap, cellArray);
+        cellArray_ = std::make_shared<CellArray>(
+          cellArray, cellNumber, cellArray[0] - 1);
+      }
+
+      return 0;
+    }
+#endif
+
+    /**
+     * Reorder the input vertices.
+     */
+    int reorderVertices(vector<SimplexId> &vertexMap) {
+      // get the number of nodes (the max value in the array)
+      for(SimplexId vid = 0; vid < vertexNumber_; vid++) {
+        if(vertexIndices_[vid] > nodeNumber_) {
+          nodeNumber_ = vertexIndices_[vid];
+        }
+      }
+      nodeNumber_++; // since the index starts from 0
+      vector<vector<SimplexId>> nodeVertices(nodeNumber_);
+      for(SimplexId vid = 0; vid < vertexNumber_; vid++) {
+        nodeVertices[vertexIndices_[vid]].push_back(vid);
+      }
+
+      // update the vertex intervals
+      vertexIntervals_.resize(nodeNumber_ + 1);
+      vertexIntervals_[0] = -1;
+      SimplexId vertexCount = 0;
+      for(SimplexId nid = 0; nid < nodeNumber_; nid++) {
+        for(SimplexId vid : nodeVertices[nid]) {
+          vertexMap[vid] = vertexCount++;
+        }
+        vertexIntervals_[nid + 1] = vertexCount - 1;
+      }
+
+      // rearange the vertex coordinate values
+      if(doublePrecision_) {
+        double *newPointSet = new double[3 * vertexNumber_];
```

Comment:
again, the usage of `new` and `delete` is strongly discouraged.
in this instance `vector<double> newPointSet(3*vertexNumber_);` seems to do the trick.

Reply:
solved